src/rust/engine/src/nodes.rs

```diff
@@ -381,6 +409,76 @@ impl WrappedNode for Select {
   }
 }
 
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub struct HttpRequester(Key);
+
+impl From<HttpRequester> for NodeKey {
+  fn from(n: HttpRequester) -> Self {
+    NodeKey::HttpRequester(n)
+  }
+}
+
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct EngineHttpResponse {
+  response_code: Option<u16>,
+  bytes: Option<Vec<u8>>,
+}
+
+impl WrappedNode for HttpRequester {
+  type Item = EngineHttpResponse;
+
+  fn run(self, context: Context) -> NodeFuture<EngineHttpResponse> {
+    #[allow(unused_imports)]
+    let value = externs::val_for(&self.0);
+    let url_string = externs::project_str(&value, "url");
+    let url = try_future!(Url::parse(&url_string).map_err(|err| throw(&format!(
+      "HttpRequest - error parsing URL {}: {}",
+      url_string, err
+    ))));
+
+    let core = context.core.clone();
+    let send_future = core.http_client.get(url).send();
+
+    //The Ok variant of this result is reqwest::Response, but for some reason actually importing
+    //that and trying to use it here breaks, even though it doesn't break with reqwest::Error
```

Comment:
Could we be much more specific about what “breaks” here?

Reply:
Also, the style I’ve seen everywhere in Rust and elsewhere has a single space after a `//` comment. Most text editors will have a button to do this automatically so you don’t have to type it out by hand each time!

Reply:
Also, I know I haven’t always done this, but I would like to have comments end in periods (or exclamation points!). This is already a great comment, so just adding the period is sufficient here I think.

Reply:
Found a way to specify the type annotation correctly (I think I was just looking at the documentation for a different version of reqwests than we're using)